runtime-kernel/rpi-firmware-boot/autobuild/postinst

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+if [ -d "/tree/runtime-kernel" ]; then
+  echo "Running in build environment, skipping firmware update."
+  exit 0
+fi
+
+if ! grep -q "/boot/rpi" /etc/fstab; then
+  echo "Mount point '/boot/rpi' does not exist in fstab file."
+  echo "You need to update firmware files manually, which is installed to /usr/lib/rpi64/boot."
+  echo "Or add the boot partition to fstab and try again."
+  exit 1
+fi
+
+if ! mount -a; then
+  # echo "Unable to mount /boot/rpi! Please check your fstab file. Not treating as an error, exiting cleanly."
+  echo "Failed to mount /boot/rpi! Please check your fstab file."
+  echo "Your firmware is not updated."
+  exit 1
+fi
+
+if [ -e /boot/rpi/kernel8.img ]; then
+  echo "Copying new firmware..."
+  cp -r /usr/lib/rpi64/boot/* /boot/rpi/
+elif [ -e /boot/rpi/kernel_2712.img ]; then
+  echo "Copying new firmware..."
+  cp -r /usr/lib/rpi64/boot/* /boot/rpi/
+else
+  echo "Copying kernel and firmware..."
+  cp -r /usr/lib/rpi64/boot/* /boot/rpi/
+  if [ -e /usr/lib/rpi64/kernel/kernel8.img ]; then
+    cp -r /usr/lib/rpi64/kernel/* /boot/rpi/
+  elif [ -e /usr/lib/rpi64/kernel/kernel_2712.img ]; then
+    cp -r /usr/lib/rpi64/kernel/* /boot/rpi/
+  fi
+fi
```

Comment:
Since we have to put the kernels in their separate directories and we can not know the version of the kernel installed on the system, so I think this entire logic can be simplified to install the firmware only and unconditionally:

```bash
echo "Installing the boot firmware ..."
cp -r /usr/lib/rpi64/boot/* /boot/rpi/
```